docs/querying/query-context-reference.md

```diff
@@ -44,6 +44,7 @@ Unless otherwise noted, the following parameters apply to all query types, and t
 |Parameter          |Default                                 | Description          |
 |-------------------|----------------------------------------|----------------------|
 |`timeout`          | `druid.server.http.defaultQueryTimeout`| Query timeout in millis, beyond which unfinished queries will be cancelled. 0 timeout means `no timeout` (up to the server-side maximum query timeout, `druid.server.http.maxQueryTimeout`). To set the default timeout and maximum timeout, see [Broker configuration](../configuration/index.md#broker) |
+|`perSegmentTimeout`| `null`                                 | Per-segment processing timeout in millis, beyond which unfinished queries will be cancelled. Should be ≤ `timeout`. 0 `perSegmentTimeout` means `no per-segment timeout`. Generally, a standard default should be O(X seconds). A cluster-wide default value for this query context can be specified via `druid.query.default.context.perSegmentTimeout`.|
```

Comment:
> Should be ≤ `timeout`

I don't think there's any validation on the query path that checks for this invariant right? So this is more of a recommendation?